### DISCUSSION TITLE
Design suggestions for the mods' customisations

### DISCUSSION THREAD

**Discussion by Gabixel** [ppy/osu#18507] [Ideas]
Currently, the mod customisation window is something like this:

![osu_2022-05-31_13-04-58](https://user-images.githubusercontent.com/43073074/171165988-996bc4f2-9033-4c54-8c8e-4eab49aba774.png)

It can be not very pleasant to scroll horizontally with multiple mods selected.

I know that there's going to be a redesign with a drop-down/side menu (I think), but I would like to suggest my idea to view the mod customisation more appreciably and rapidly. Here's a "quick" example I've made in Figma:

![mod customisation](https://user-images.githubusercontent.com/43073074/171166748-133695ae-3cc8-414d-bc83-e07eb8b3db90.png)

The idea is to show the customisation directly inside the mods' panels. This gives to the user the ability to deselect individual mods that he no longer wants to play (without changing the view). For the animation, I guess the panels can make a fade-out animation to the unselected mods (or maybe the panels can flip like a card?), or it can be simply a toggle, thus keeping all the mods visible, showing the customization only to the active ones. That's all!

**Comment by bdach** [comment]
this has been suggested by people multiple times, and my issue is that this is going to look bad with mods that have multiple settings. your proposal design conveniently mostly shows single scroll bars under the mod. i invite you to try this exercise with classic mod, or difficulty adjust, and see how it looks then.

**Reply by peppy** [reply]
We've also discussed having a popover version of controls, which will likely work better.
